src/HttpClient.ts

```diff
@@ -45,6 +45,32 @@ export class HttpClient {
         return await this._axiosInstance(requestParams);
     }
 
+    /**
+     * Performs a request with polling using the provided request parameters.
+     * @param requestParams - The request parameters.
+     * @param retries - number of retries.
+     * @param interval - The time interval in milliseconds between each retry.
+     * @returns a promise that resolves with the client response.
+     */
+    public async doRequestWithPolling(
+        requestParams: IRequestParams,
+        retries?: number,
+        interval?: number
+    ): Promise<IClientResponse> {
+        retries = retries ?? HttpClient.DEFAULT_POLLING;
+        const i: number = interval ?? HttpClient.DEFAULT_RETRY_DELAY_IN_MILLISECONDS;
+        axiosRetry(this._axiosInstance, {
+            retries: retries,
+            retryCondition: (error: AxiosError) =>
+                (error.response?.status ?? 0 >= 500) || error.response?.status === 400,
+            retryDelay: (retryCount: number) => {
+                this.logger?.debug(`Retry #${retryCount}...`);
+                return retryCount * i;
+            },
+        } as IAxiosRetryConfig);
```

Comment:
Let's reuse the axiosRetry code that in the constructor by exporting it to a new function